src/analysis/normed_space/lp_space.lean

```diff
@@ -588,6 +592,58 @@ instance : module 𝕜 (lp E p) :=
 
 @[simp] lemma coe_fn_smul (c : 𝕜) (f : lp E p) : ⇑(c • f) = c • f := rfl
 
+instance [has_smul 𝕜' 𝕜] [Π i, is_scalar_tower 𝕜' 𝕜 (E i)] :
+  is_scalar_tower 𝕜' 𝕜 (lp E p) :=
+⟨λ r c f, subtype.ext $ (lp.coe_fn_smul _ _).trans (smul_assoc _ _ _)⟩
+
+theorem _root_.nnreal.has_sum_mono {α : Type u_1} {f g : α → nnreal} {sf sg : nnreal}
+  (hf : has_sum f sf) (hg : has_sum g sg) (h : f ≤ g) :
+  sf ≤ sg :=
+begin
+  obtain rfl | hlt := eq_or_lt_of_le h,
+  { exact (hf.unique hg).le },
+  { exact (nnreal.has_sum_strict_mono hf hg hlt).le },
+end
+
+lemma norm_const_smul_le (hp : p ≠ 0) (c : 𝕜) (f : lp E p) : ‖c • f‖ ≤ ‖c‖ * ‖f‖ :=
+begin
+  rcases p.trichotomy with rfl | rfl | hp,
+  { exact absurd rfl hp },
+  { cases is_empty_or_nonempty α; resetI,
+    { simp [lp.eq_zero' f], },
+    -- surely I'm missing some `is_lub` API here.
+    have hcf := lp.is_lub_norm (c • f),
+    have hfc := (lp.is_lub_norm f).mul_left (norm_nonneg c),
+    simp_rw [←set.range_comp, function.comp] at hfc,
+    refine hcf.right _,
+    have := hfc.left,
+    simp_rw [mem_upper_bounds, set.mem_range, forall_exists_index, forall_apply_eq_imp_iff'] at this ⊢,
+    intro a,
+    refine (norm_smul_le _ _).trans (this a) },
```

Comment:
@YaelDillies, do you know what the API I'm looking for here is? I want to say "`is_lub (range f1) b1` and `is_lub (range f2) b2` implies `b1 ≤ b2` if `f1 ≤ f2`".

Reply:
I've tweaked the comment slightly. I don't think there's anything mathematically interesting here: so I think it's ok to have a lousy proof and clean it up in mathlib4. This file is now on the porting shore, so this PR is running out of time.